### PR TITLE
feat: jq support

### PR DESCRIPTION
**Pull request by ssddOnTop**
Issue Reference(s):
Fixes https://github.com/tailcallhq/tailcall/issues/1321
/claim #1321

## Summary by CodeRabbit

- **New Features**
	- Enhanced the GraphQL schema to simplify the user query argument structure.
	- Introduced new evaluation methods across various modules to enhance processing capabilities.
	- Updated mustache template syntax for improved clarity and consistency in variable references.
  
- **Bug Fixes**
	- Adjusted key-value pair transformations to ensure correct string conversion in headers management.
  
- **Documentation**
	- Updated internal test cases to align with the new mustache syntax and evaluation methods.

### CODE REVIEW

File: src/path.rs [hunk 1]
```diff
@@ -31,6 +33,17 @@ impl PathString for serde_json::Value {
             _ => Cow::Owned(a.to_string()),
         })
     }
+
+    fn evaluate(&self, filter: &Filter) -> Option<Cow<'_, str>> {
```
Comment (amitksingh1490) [inline src/path.rs:37]: Should return a Value instead of string

File: examples/jsonplaceholder.graphql [hunk 2]
```diff
@@ -24,5 +24,5 @@ type Post {
   userId: Int!
   title: String!
   body: String!
-  user: User @call(steps: [{query: "user", args: {id: "{{.value.userId}}"}}])
+  user: User @call(steps: [{query: "user", args: {id: "{{.userId}}"}}])
```
Comment (tusharmath) [inline examples/jsonplaceholder.graphql:27]: We shouldn't need this change.

File: src/path.rs [hunk 3]
```diff
@@ -31,6 +33,18 @@ impl PathString for serde_json::Value {
             _ => Cow::Owned(a.to_string()),
         })
     }
+
+    fn evaluate(&self, filter: &Filter) -> Option<async_graphql::Value> {
```
Comment (tusharmath) [inline src/path.rs:37]: Move this function to `impl Mustache` Should take in a `async_graphql::Value` and produce as `async_graphql::Value`

File: src/serde_value_ext.rs [hunk 4]
```diff
@@ -71,7 +71,7 @@ mod tests {
 
     #[test]
     fn test_render_value_nested_str() {
-        let value = json!({"a": "{{foo.bar.baz}}"});
```
Comment (tusharmath) [inline src/serde_value_ext.rs:74]: These test should have failed when we migrated to jq syntax, do u know why it did not fail?

File: src/mustache.rs [hunk 5]
```diff
@@ -1,17 +1,22 @@
...
+use serde::Serialize;
 
 use crate::path::{PathGraphql, PathString};
 
-#[derive(Debug, Clone, PartialEq, Hash)]
-pub struct Mustache(Vec<Segment>);
+#[derive(Debug, Clone)]
+pub struct Mustache {
+    segments: Vec<Segment>,
+    jacques: Option<jaq_interpret::Filter>,
```
Comment (tusharmath) [inline src/mustache.rs:18]: Should be an enum.

**Pull request closed without merging**